choreolib/src/main/java/choreo/auto/AutoTrajectory.java

```diff
@@ -127,7 +127,7 @@ <SampleType extends TrajectorySample<SampleType>> AutoTrajectory(
    * Returns true if alliance flipping is enabled and the alliance optional is present. Also returns
    * true if alliance flipping is disabled.
    */
-  private boolean allowSampling() {
+  private boolean allianceKnownOrIgnored() {
     return routine.allianceKnownOrIgnored.getAsBoolean();
```

Comment:
I'm not a fan of pulling protected variables from other classes, this should be injected in the constructor rather than pulling from `AutoRoutine`. Doing so would also resolve my comment about `VOID-ROUTINE` forcing `allianceKnownOrIgnored` to `true`. Consequently, protected fields of `AutoRoutine` should be made private.

Note that `routine.isActive` is used a lot in `AutoTrajectory`, which should be replaced with `routine.active().getAsBoolean()`.